able to build and start dockerized rpc adaptor

shunjizhan opened the pull request:
## Change
Added Dockerfile to build adn run `eth-rpc-adaptor` service.

Note that this uses `yarn` for itself, instead of `rush` for monorepo, so it needs a slightly different `package.json`, and a `yarn.lock` just for docker (rush won't need to use it)

## Test
- able to start a dockerized `eth-rpc-adaptor` service, and connected to the latest public mandala testnet.
- tested a couple `eth_xxx` and they worked.

## TODO
currently any `eth_xxx` that interacts with evm-subql won't work, we will need to add support for it after we have a public evm-subql service.

xlc reviewed (commented):
better to have ./docker/package.json and ./docker/yarn.lock so it is easier to update them and the IDE understand the types

xlc commented on eth-rpc-adapter/Dockerfile:
```diff
@@ -0,0 +1,21 @@
+FROM node:14-alpine
+
+ARG mandala_url=wss://node-6870830370282213376.rz.onfinality.io/ws?apikey=0f273197-e4d5-45e2-b23e-03b015cb7000
```
No need to inject this at build time. We can configure environment variable at deployment time.

shunjizhan commented:
> better to have ./docker/package.json and ./docker/yarn.lock so it is easier to update them and the IDE understand the types

moved

> No need to inject this at build time. We can configure environment variable at deployment time.

removed, now uses local `.env` file, same as running locally

tj4laminar commented on eth-rpc-adapter/Dockerfile:
```diff
@@ -0,0 +1,16 @@
+FROM node:14-alpine
+
```
is it current latest version of node-alpine image ?

shunjizhan replied:
latest version is 17 but we use 14 across `bodhi.js` project

tj4laminar commented on eth-rpc-adapter/Dockerfile:
```diff
@@ -0,0 +1,16 @@
+FROM node:14-alpine
+
+ENV NODE_ENV=production
+
```
is it required field or just get as it was in example code which shared by Bryan ?

xlc replied:
we want to have this for prod image

tj4laminar commented on eth-rpc-adapter/Dockerfile:
```diff
@@ -0,0 +1,16 @@
+RUN [ "yarn", "build" ]
+
+EXPOSE 8545
+
```
is it kind of incoming port or just connecting to external world ?

xlc replied:
this will be exposed to external world

tj4laminar commented on eth-rpc-adapter/Dockerfile:
```diff
@@ -0,0 +1,16 @@
+
+RUN yarn
+RUN [ "yarn", "build" ]
+
```
let run this container as non-root user

shunjizhan replied:
now use user 'node'

tj4laminar reviewed (approved):
looks good
